scipy-1.0/paper.tex

```diff
@@ -296,6 +300,25 @@ \subsubsection*{LowLevelCallable}
 
 \subsection*{Cython bindings for BLAS, LAPACK and special}
 
+When writing algorithms or wrappers in Cython\cite{behnel2011cython}, it is often convenient to call directly into the libraries that SciPy wraps rather than routing the call back through the Python interpreter.
+SciPy currently includes Cython wrappers\cite{blas-lapack-wrappers-scipy-2015} for many BLAS and LAPACK\cite{LAPACK} routines and for many of the special functions provided in the \texttt{scipy.{\allowbreak}special} submodule.
+These wrappers are available in the modules \texttt{scipy.{\allowbreak}linalg.{\allowbreak}cython\_blas}, \texttt{scipy.{\allowbreak}linalg.{\allowbreak}cython\_lapack}, and \texttt{scipy.{\allowbreak}special.{\allowbreak}cython\_special} respectively.
+The low-level interfaces for Cython can also be used by third party libraries to get direct access to the functions in these libraries while avoiding the overhead of Python function calls{\textemdash}this can give performance gains of one or two orders of magnitude for many use cases.
+Cython's cimport machinery allows Cython based modules to share C-level objects with other Cython modules without requiring that the third-party extensions link against any additional libraries.
+Function pointers to wrappers for library routines can also be shared in this way, so that Cython modules using the interface provided by SciPy no longer have to worry about finding and linking against the original library.
+This lets other extensions avoid the complexity of finding and using the correct libraries on the various possible Fortran compiler ABI and OS combinations.
+The low-level Cython wrappers provide an OS and Fortran compiler independent interface for the libraries they wrap.
```

Comment:
Perhaps you are making a few significant, but only subtly distinct, points in the preceeding four sentences (307-310), but can't distinguish among them and am somewhat confused. 

I think the central point is that these Cython wrappers allow the use of BLAS, LAPACK, and special functions 1) efficiently (as mentioned before) and 2) conveniently, avoiding complicated linking stuff.

Can this be simplified?

Reply:
Yah, this one can be simplified. The idea I was expressing is that cimporting these wrappers relies on Cython shuttling a bunch of function pointers around through the Python interpreter instead of just linking against the original library. Strictly speaking, that's not the only way to set up Cython wrappers, but the level of detail there obscures the main points about why the wrappers are useful in the first place.